foursight_core/react/api/checks.py

```diff
@@ -31,50 +33,119 @@ def get_checks_raw(self) -> dict:
     def get_checks(self, env: str) -> dict:
         """
         Returns a dictionary containing all checks, annotated with various info,
-        e.g. the (cron) schedule from the associated lambdas, and check kwargs,
-        from the check function decorators; filtered by the given env name.
-        Cached on/after the first call.
+        e.g. the (cron) schedule from the associated lambdas, check kwargs, and
+        associated actions, from the check function decorators; filtered by the
+        given env name. Cached on the first call, except for the filtering part.
         """
         if not Checks._cached_checks:
             checks = self._check_setup
             for check_key in checks.keys():
                 checks[check_key]["name"] = check_key
                 checks[check_key]["group"] = checks[check_key]["group"]
             lambdas = self.get_annotated_lambdas()
+            self._annotate_checks_for_dependencies(checks)
             self._annotate_checks_with_schedules_from_lambdas(checks, lambdas)
             self._annotate_checks_with_kwargs_from_decorators(checks)
             Checks._cached_checks = checks
         return self._filter_checks_by_env(Checks._cached_checks, env)
 
+    @memoize
     def get_checks_grouped(self, env: str) -> list:
         """
         Like get_checks but returns the checks grouped by their group names.
         """
-        checks_groups = []
+        grouped_checks = []
         checks = self.get_checks(env)
         for check_name in checks:
             check_item = checks[check_name]
             check_item_group = check_item["group"]
-            # TODO: Probably a more pythonic way to do this.
             found = False
-            for grouped_check in checks_groups:
+            for grouped_check in grouped_checks:
                 if grouped_check["group"] == check_item_group:
                     grouped_check["checks"].append(check_item)
                     found = True
                     break
             if not found:
-                checks_groups.append({"group": check_item_group, "checks": [check_item]})
-        return checks_groups
+                grouped_checks.append({"group": check_item_group, "checks": [check_item]})
+        return grouped_checks
+
+    @memoize
+    def get_checks_grouped_by_schedule(self, env: str) -> list:
+        """
+        Like get_checks_grouped but groups by schedule (i.e. by scheduling lambdas).
+        """
+        def create_schedule_group_title(name: str):
+            """
+            Pretty lame title-ifying the groiup name for display purposes, based
+            on what we knnow about the currently existing scheduling lambda names.
+            """
+            name = name.replace("Min", " Minute")
+            if name.endswith("Checks"):
+                return name[:len(name) - len("Checks") - 0]
+            elif name.endswith("Checks1"):
+                return name[:len(name) - len("Checks1") - 0] + " I"
```

Comment:
You can probably do this more simply with a replace operation based on the number? It's also not immediately obvious to me why this isn't done in the front-end.